kernel/src/process_load_utilities.rs

```diff
@@ -0,0 +1,482 @@
+    /// - `4`: Initialize proc_data.dynamic_unsued_sram_start_addr with sram_end_address returned from load_processes_advanced
+    ///        This initial value comes from the result value of 'kernel::process::load_processes' at main.rs (set only one time at OTA_app init stage)
+    ///        This inital value is copied to internal grant variable, and this grant variable is used in 'fn load_processes_advanced_air' and updated after loading an application
+    ///        We don't have to interrupt the sram region already used by kernel and other apps
+    /// - `5`: Calculate CRC32-POXIS of the flashed app region and return the result value
+    /// - `6`: Return the supported maximum process number by platform
+    /// - `7`: Return the end address of flash memory (i.e., 0x80000 in case of this platform)
+    /// - `8`: Return an index that is used to store the entry point of an app flashed into PROCESS global array
+    ///        With this index, we prevent the kernel from loading 4 more than applications
+    /// - `9`: Return the start address of flash memory allocated to apps (i.e., 0x40000 in case of this platform)
+    
+    fn command(
+        &self,
+        command_num: usize,
+        _unused1: usize,
+        _unused2: usize,
+        appid: ProcessId,
+    ) -> CommandReturn {
+        match command_num {
+            0 => CommandReturn::success(),
+            
+            1 =>
+            /* perform load process work */
+            {
+                let res = self.data.enter(appid, |proc_data, _| {
+                    self.load_processes_air(proc_data)
+                })
+                .map_err(ErrorCode::from);
+        
+                match res {
+                    Ok(Ok(())) => CommandReturn::success(),
+                    Ok(Err(e)) => CommandReturn::failure(e),
+                    Err(e) => CommandReturn::failure(e),
+                }
+            }
+
+            2 =>
+            /* find dynamically changing start address of writable flash memory immediately next to the last application already loaded */
+            {   
+                let res = self.data.enter(appid, |proc_data, _| {
+                    self.find_dynamic_start_address_of_writable_flash(proc_data)
+                })
+                .map_err(ErrorCode::from);
+        
+                match res {
+                    Ok(Ok(())) => CommandReturn::success(),
+                    Ok(Err(e)) => CommandReturn::failure(e),
+                    Err(e) => CommandReturn::failure(e),
+                }
+            }
+
+            3 =>
+            /* Return the dynamically changing start address after commnad 2 */
+            {
+                self.data.enter(appid, |proc_data, _| {
+                    CommandReturn::success_u32(proc_data.dynamic_flash_start_addr as u32)
+                })
+                .unwrap_or(CommandReturn::failure(ErrorCode::FAIL))
+            }
```

Comment:
What is this API used for?

Reply:
Command 3 API is used in ota_app in order to check whether or not there is enough flash region before writing binary data. (e.g., 0x80000 - proc_data.dynamic_flash_start_add > app size). Command 3 is called after calling Command 2 from ota_app.